### ISSUE TITLE
Console Error (OTSessionErrorDomain error 1113) when Unmounting OT Components

### ISSUE THREAD

**Issue by jonathanwmaddison** [opentok/opentok-react-native#18]
Thanks for this awesome new library! Very clean and so far I have it working on iOS.

I noticed on iOS that I receive an error anytime I unmount the Video component which contains the OTSession, Publisher and Subscriber components.

I don't have any significant event handlers or anything that I think would be causing an issue.
This is the specific error message that's logged in my React Native Debugger, after unmounting:
<img width="445" alt="screen shot 2018-04-12 at 3 30 52 pm" src="https://user-images.githubusercontent.com/22892848/38699831-320eaca6-3e67-11e8-9931-00f8b67b37b9.png">

Is there anything I should do before I unmount the my component containing the OT components?

Here is the abbreviated JSX that I'm unmounting, leading to the error:

```js
class Video extends Component ....

handleStreamEvent = () => {
    this.startTimer();
    this.setState(({ streamActive }) => ({
      streamActive: !streamActive,
    }));
  };

  sessionEventHandlers = {
    streamCreated: this.handleStreamEvent,
    streamDestroyed: this.handleStreamEvent,
  };
    //
          <OTSession
            sessionId={sessionId}
            apiKey={tokBoxApiKey.toString()}
            token={token}
            style={{ flex: 1 }}
            eventHandlers={this.sessionEventHandlers}
          >
            <OTSubscriber style={{ width: '100%', height: '100%' }} />
            <OTPublisher
              style={styles.publisherContainer}
              properties={{
                publishAudio,
                publishVideo,
              }}
            />
          </OTSession>

```

**Comment by jonathanwmaddison** [comment]
From the OT docs this is what domain error 1113 refers to:

> OTSessionPublisherNotFound
> The publisher is unknown to this session. This is usually the result of attempting to unpublish a publisher that is not associated with the session.

https://tokbox.com/developer/sdks/ios/reference/Constants/OTSessionErrorCode.html

**Comment by msach22** [comment]
@jonathanwmaddison Thanks for reporting this! Looking at the error, it looks like the session gets disconnected first and then the publisher trying to unpublish, but since there is no session, the unpublish fails. You should unmount the `OTPublisher` before unmounting the `OTSession` component.

Regardless, we can implement a check in the native code to see if the session is still connected or not before invoking the `unpublish` method. I also noticed an issue with `setState` that I have in the `OTSession` component on unmounting so I'll send a PR to fix that shortly.

**Comment by msach22** [comment]
@jonathanwmaddison #22 should take care of this for you!

**Comment by msach22** [comment]
Just released `v0.4.0` which includes the feature that checks for session connection on both iOS and Android and patch for the `setState` that **was** called in `componentWillUnMount` of `OTSession`

**Comment by jonathanwmaddison** [comment]
@msach22 thank you! 🏅 💯
